plotly/tools.py

```diff
@@ -398,60 +399,327 @@ def get_subplots(rows=1, columns=1, horizontal_spacing=0.1,
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], xaxis='x3', yaxis='y3')]
+
+    Example 4:
         # print out string showing the subplot grid you've put in the layout
         fig = tools.get_subplots(rows=3, columns=2, print_grid=True)
 
-    key (types, default=default):
-        description.
+    Keywords arguments with constant defaults:
+
+    rows (kwarg, int, default=1):
+        Number of rows on the figure.
+
+    columns (kwarg, int, default=1):
+        Number of columns on the figure.
+
+    shared_xaxes (kwarg, boolean or list, default=False)
+        Assign shared x axes.
+        If True, all x axes are shared.
+        To assign shared x axes per subplot grid cell (see 'specs'),
+        send list (or list of lists, one list per shared axis)
+        of cell index tuples.
+
+    shared_yaxes (kwarg, boolean or list, default=False)
+        Assign shared y axes.
+        If True, all y axes are shared.
+        To assign shared y axes per subplot grid cell (see 'specs'),
+        send list (or list of lists, one list per shared axis)
+        of cell index tuples.
+
+    print_grid (kwarg, boolean, default=False):
+        If True, prints a tab-delimited string representation of
+        your plot grid.
 
-    rows (int, default=1):
-        Number of rows, evenly spaced vertically on the figure.
+    Keyword arguments with variable defaults:
 
-    columns (int, default=1):
-        Number of columns, evenly spaced horizontally on the figure.
+    horizontal_spacing (kwarg, float in [0,1], default=0.2 / columns):
+        Space between subplot columns.
 
-    horizontal_spacing (float in [0,1], default=0.1):
-        Space between subplot columns. Applied to all columns.
+        Applies to all columns (use 'specs' subplot-dependents spacing)
 
-    vertical_spacing (float in [0,1], default=0.05):
-        Space between subplot rows. Applied to all rows.
+    vertical_spacing (kwarg, float in [0,1], default=0.3 / rows):
+        Space between subplot rows.
+        Applies to all rows (use 'specs' subplot-dependents spacing)
 
-    print_grid (True | False, default=False):
-        If True, prints a tab-delimited string representation of your plot grid.
+    specs (kwarg, list (of lists) of dictionaries):
+        Subplot specifications.
 
+        - Indices of the outer list correspond to subplot grid rows
+          starting from the bottom.
+
+        - Indices of the inner lists correspond to subplot grid columns
+          starting from the left
+
+        - Note that specs[0][0] has the specs for the bottom-left subplot
+
+        - Each item in the 'specs' list corresponds to one subplot
+          in a subplot grid. The subplot grid has 'rows' times 'columns'
+          cells.
+
+        - Each item in the 'specs' is a dictionary.
+            The available keys are:
+
+            * isEmpty (boolean, default=False): flag for empty grid cells
+            * is3D (boolean, default=False): flag for 3d scenes
```

Comment:
Good call about `is_empty` and `is_3d`  (I've been written a lot of js lately, sorry)

Matplotlib's gridspec uses `rowspan` and `colspan`, so I would vote for keeping them that way.